packages/vite/src/plugins/plugin.ts

```diff
@@ -237,16 +241,28 @@ function getOutputs(
 
 function normalizeOutputPath(
   outputPath: string | undefined,
-  projectRoot: string
+  projectRoot: string,
+  path: 'coverage' | 'dist'
 ): string | undefined {
-  if (!outputPath) return undefined;
-  if (isAbsolute(outputPath)) {
-    return `{workspaceRoot}/${relative(workspaceRoot, outputPath)}`;
+  if (!outputPath) {
+    if (projectRoot === '.') {
+      return `{projectRoot}/${path}`;
+    } else {
+      return `{workspaceRoot}/${path}/{projectRoot}`;
+    }
   } else {
-    if (outputPath.startsWith('..')) {
-      return join('{workspaceRoot}', join(projectRoot, outputPath));
+    if (isAbsolute(outputPath)) {
+      return `{workspaceRoot}/${relative(workspaceRoot, outputPath)}`;
     } else {
-      return outputPath;
+      if (outputPath.startsWith('..')) {
+        return join('{workspaceRoot}', join(projectRoot, outputPath));
+      } else {
+        if (projectRoot === '.' && outputPath.startsWith('./')) {
```

Comment:
I think this if-else can just be:

```
join('{projectRoot}', outputPath.slice(2));
```

since joining `./foo` or `foo` gives the same result.

Reply:
ohhhhhhhh 😅